plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -198,8 +198,44 @@ public function __construct( $args = array() ) {
 		// TODO: args to be implemented.
 		unset( $this->args['customer_note'], $this->args['name'] );
 
+		$require_limit_data = ( $this->arg_isset( 'limit' ) || $this->arg_isset( 'page' ) || $this->arg_isset( 'offset' ) );
+
 		$this->build_query();
-		$this->run_query();
+
+		/**
+		 * Filters the orders array before the query takes place.
+		 *
+		 * Return a non-null value to bypass the HPOS default order queries.
+		 *
+		 * If the query includes limits via the `limit`, `page`, or `offset` arguments, we
+		 * require the `found_orders` and `max_num_pages` properties to also be set.
+		 *
+		 * @since 8.2.0
+		 *
+		 * @param array $order_data {
+		 *     An array of order data.
```

Comment:
```suggestion
		 * @param array|null $order_data {
		 *     An array of order data, or null if the default query should run.
		 *
```

☝🏼 I wonder if this might be a little more accurate, and also closer to WP norms for `pre_*`-style filters:

- Callbacks will receive either an array or null, whereas this currently suggests they will always receive an array. Sidenote: could also use `array?` instead of `array|null`.
- At least to my mind, it makes it a little clearer that they can simply return null and not something like `[ null, ... ]` if they do not want to short-circuit things.

Reply:
@barryhughes if you look at the whole comment (with the sql one added): 

```
		 * @param array $order_data {
		 *     An array of order data.
		 *     @type int[] $orders        Return an array of order IDs data to short-circuit the HPOS query,
		 *                                or null to allow HPOS to run its normal query.
		 *     @type int   $found_orders  The number of orders found.
		 *     @type int   $max_num_pages The number of pages.
		 * }
		 * @param OrdersTableQuery   $query The OrdersTableQuery instance.
		 * @param string             $sql The OrdersTableQuery instance.
``` 

The first @param will always be an array, the actual order IDs will be the first item (possibly null), and the other two will be ints about those orders. I'm adding a test to ensure that this construct can also be null. Does that sound alright? 

(This first param is also the thing that will be returned by the filter, and with that one we had to go with such a construct to be able to return multiple values.)

Reply:
> The first @param will always be an array

Well, by default, the value of the first param is *null*—so, at the risk of sounding pedantic, it will not always be an array. Also, leaving the value as *null* is a valid choice for any callbacks that are registered (they may only transform it to an array if some condition is met).

I think what I'm proposing is the more accurate description of the param, but, you can treat this as optional (... and, I think you're right: if the entire comment is read, things are sufficiently clear).

Reply:
> by default, the value of the first param is null—so, at the risk of sounding pedantic, it will not always be an array

Thanks for catching that, of course you're absolutely right. And I don't think that's pedantic, I think we should be exact in our words so thanks a lot for that. ☺️